lib/phoenix/router.ex

```diff
@@ -537,6 +538,74 @@ defmodule Phoenix.Router do
     end
   end
 
+  defp plug_with_routes() do
+    quote location: :keep, generated: true do
+      @behaviour Phoenix.Router.PlugWithRoutes
+
+      def formatted_routes(_) do
+        __routes__()
+        |> Enum.map(fn route ->
+          Map.take(route, [
+            :verb,
+            :path,
+            :plug,
+            :plug_opts,
+            :metadata,
+            :helper
+          ])
+        end)
+      end
+
+      def verified_route?(_, split_path) do
+        case __verify_route__(split_path) do
```

Comment:
We should keep the amount of generated code to a minimum. We should probably delegate this to a `Phoenix.Router.__verified_route__` or similar.